include/verilated_trace.h

```diff
@@ -151,34 +194,86 @@ template <class T_Derived> class VerilatedTrace {
     void fullFloat(vluint32_t* oldp, float newval);
     void fullDouble(vluint32_t* oldp, double newval);
 
-    // Check previous dumped value of signal. If changed, then emit trace entry
+#ifdef VL_TRACE_THREADED
+    // Threaded tracing. Just dump everything in the trace buffer
     inline void chgBit(vluint32_t* oldp, vluint32_t newval) {
+        m_traceBufferWritep[0].cmd = VerilatedTraceCommand::CHG_BIT | newval;
+        m_traceBufferWritep[1].oldp = oldp;
+        m_traceBufferWritep += 2;
```

Comment:
Maybe I missed it some other spot, but please have a VL_DEBUG only assert that we don't write past the end of the buffer (due to e.g. allocation or other bug).

Reply:
P.S. Maybe have a "increment" macro that does the increment and check?